tests/testthat/test-qenv_constructor.R

```diff
@@ -0,0 +1,82 @@
+testthat::test_that("constructor returns qenv if nothing is specified", {
+  q <- new_qenv()
+  testthat::expect_s4_class(q, "qenv")
+  testthat::expect_identical(ls(q@env), character(0))
+  testthat::expect_identical(q@code, expression())
+  testthat::expect_identical(q@id, integer(0))
+  testthat::expect_identical(q@warnings, character(0))
+  testthat::expect_identical(q@messages, character(0))
+})
+
+testthat::test_that("parent of qenv environment is the parent of .GlobalEnv", {
+  a <- 1L
+  q1 <- new_qenv(quote(a <- 1), env = environment())
+  q2 <- new_qenv()
+  q3 <- new_qenv(code = quote(a <- 1), env = list2env(list(a = 1)))
+  testthat::expect_identical(parent.env(q1@env), parent.env(.GlobalEnv))
+  testthat::expect_identical(parent.env(q1@env), parent.env(q2@env))
+  testthat::expect_identical(parent.env(q1@env), parent.env(q3@env))
+})
+
+testthat::test_that("provided environment is copied so the change of the environment won't affect qenv", {
+  a <- 1L
+  q1 <- new_qenv(quote(a <- 1), env = environment())
+  b <- 2L
```

Comment:
should this be @gogonzo 
```suggestion
  a <- 2L
```